gnucash/register/register-gnome/combocell-gnome.c

```diff
@@ -534,37 +555,99 @@ gnc_combo_cell_modify_verify (BasicCell *_cell,
     }
 
     /* If deleting, just accept */
-    if (change == NULL)
+    if (change == NULL && !type_ahead_search)
     {
         gnc_basic_cell_set_value_internal (_cell, newval);
         return;
     }
 
     /* If we are inserting in the middle, just accept */
-    if (*cursor_position < _cell->value_chars)
+    if (*cursor_position < _cell->value_chars && !type_ahead_search)
     {
         gnc_basic_cell_set_value_internal (_cell, newval);
         return;
     }
-
+    // Legacy account name match.
```

Comment:
This is confusing because it seems to introduce a large block of mostly new code. What part is "legacy"?